pastis/launchers/run_luvoir_efields.py

```diff
@@ -12,15 +17,51 @@
     fpath = CONFIG_PASTIS.get('LUVOIR', 'harris_data_path')  # path to Harris spreadsheet
     pad_orientations = np.pi / 2 * np.ones(120)
 
-    DM_SPEC = (fpath, pad_orientations, False, True, False)
+    DM_SPEC = (fpath, pad_orientations, True, False, False)
     # DM_SPEC = tuple or int, specification for the used DM -
     #    for seg_mirror: int, number of local Zernike modes on each segment
-    #    for harris_seg_mirror: tuple (string, array, bool, bool, bool), absolute path to Harris spreadsheet, pad orientations, choice of Harris mode sets (thermal, mechanical, other)
+    #    for harris_seg_mirror: tuple (string, array, bool, bool, bool),
+    #    absolute path to Harris spreadsheet, pad orientations, choice of Harris mode sets (thermal, mechanical, other)
     #    for zernike_mirror: int, number of global Zernikes
 
     # First generate a couple of matrices
     run_matrix = MatrixEfieldLuvoirA(which_dm=DM, dm_spec=DM_SPEC, design=APLC_DESIGN,
-                                     initial_path=CONFIG_PASTIS.get('local', 'local_data_path'))
+                                     calc_science=True, calc_wfs=True,
+                                     initial_path=CONFIG_PASTIS.get('local', 'local_data_path'),
+                                     norm_one_photon=True)
     run_matrix.calc()
     dir_run = run_matrix.overall_dir
     print(f'All saved to {dir_run}.')
+
+
+    # Instantiate the LUVOIR telescope
+    optics_input = os.path.join(util.find_repo_location(), CONFIG_PASTIS.get('LUVOIR', 'optics_path_in_repo'))
+    coronagraph_design = CONFIG_PASTIS.get('LUVOIR', 'coronagraph_design')
+    sampling = CONFIG_PASTIS.getfloat('LUVOIR', 'sampling')
+    tel = LuvoirA_APLC(optics_input, coronagraph_design, sampling)
+    nm_aber = CONFIG_PASTIS.getfloat('LUVOIR', 'calibration_aberration') * 1e-9
+
+    # Create harris deformable mirror
+    tel.create_segmented_harris_mirror(fpath, pad_orientations, thermal=True, mechanical=False, other=False)
+    tel.harris_sm
+
+    # get number of poking modes
+    num_actuators = tel.harris_sm.num_actuators
+    num_modes = 5
+
+    # calculate dark hole contrast
+    tel.harris_sm.flatten()
+    unaberrated_coro_psf, ref = tel.calc_psf(ref=True, display_intermediate=False, norm_one_photon=True)
+    norm = np.max(ref)
+    dh_intensity = (unaberrated_coro_psf / norm) * tel.dh_mask
+    contrast_floor = np.mean(dh_intensity[np.where(tel.dh_mask != 0)])
+    print(f'contrast floor: {contrast_floor}')
```

Comment:
If you run the analysis above, all of this has already been done and is either accessible within the class or saved to disk.

Reply:
@ivalaginja can you please point out how to access this?
I used util.dh_mean(im, dh) to get the contrast,
where im = fits.getdata(os.path.join(dir_run, 'unaberrated_coro_psf.fits'))
dh = run_matrix.simulator.dh_mask.shaped
And I get, contrast_floor = 5.39e-11

Whereas using the code I previously wrote, I get 4.636e-11

Reply:
There seems to be a discrepancy between util.dh and my way of calculating contrast.

Reply:
This is due to the parameter "robutness_px". While independently instantiating the simulator I took robutness_px =4.0 and computed the contrast = 4.636e-11). But when run pastis matrix is calculated, automatically robutness_px is set to 0.0. 
For future errors, I should keep in mind the way I "manually "instantiate the simulator and the way simulator is instantiated while pastis matrix run. 
Therefore those two contrast values are for different two different robustness_px. 
@ivalaginja should robustness_px be a hard coded parameter?